src/FacultyDirectory/ClientApp/src/components/UserTable/UserInput.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from 'react';
+import { Button, Input, InputGroup, InputGroupAddon } from 'reactstrap';
+import { IUser } from '../../models/IUsers';
+
+interface IProps {
+  users: IUser[];
+  onChange: Function;
```

Comment:
this is fine for this one but for more involved components try something like `onChange: (users: IUser[]) => void;` to fully specify the types.